component/client/chat/ChatHistory.tsx

```diff
@@ -88,13 +95,15 @@ class ChatHistory extends React.Component<IHistoryProps, IHistoryState> {
   }
 
   private instantiateRoomService(props: IHistoryProps): void {
+    console.log("[ INFO ] : instantiateRoomService");
     this.setState({messages: []});
 
     // Set up RoomService.
     this.roomService = new RoomService(props.url, (data: IData, user) => {
       this.setState((prevState, nextProps) => {
-        prevState.messages.push(this.createMessage(data, user));
-        return prevState;
+        const updatedMessages = prevState.messages.concat(this.createMessage(data, user));
+        console.log("[ INFO ] : messages", updatedMessages);
```

Comment:
The list of messages can be passed into ChatHistory as arguments. I should be state of ClientChat.